boule/triaxialellipsoid.py

```diff
@@ -0,0 +1,254 @@
+            )
+
+    @semimedium_axis.validator
+    def _check_semimedium_axis(
+        self, semimedium_axis, value
+    ):  # pylint: disable=no-self-use,unused-argument
+        """
+        Check if semimedium_axis is positive
+        """
+        if not value > 0:
+            raise ValueError(
+                f"Invalid semi-medium axis '{value}'. Should be greater than zero."
+            )
+        """
+        Check if semimedium axis is the middle of the 3 axes.
+        """
+        if self.semiminor_axis > value:
+            raise ValueError(
+                f"Invalid semi-minor / semi-medium axis combination.                 The semimedium axis must be larger than the semi-minor axis.                Semi-medium axis was '{value}' and the semi-minor axis was '{self.semiminor_axis}'"
+            )
+
+    @semiminor_axis.validator
+    def _check_semiminor_axis(
+        self, semiminor_axis, value
+    ):  # pylint: disable=no-self-use,unused-argument
+        """
+        Check if semiminor_axis is positive
+        """
+        if not value > 0:
+            raise ValueError(
+                f"Invalid semi-minor axis '{value}'. Should be greater than zero."
+            )
+
+    @geocentric_grav_const.validator
+    def _check_geocentric_grav_const(
+        self, geocentric_grav_const, value
+    ):  # pylint: disable=no-self-use,unused-argument
+        """
+        Warn if geocentric_grav_const is negative
+        """
+        if value < 0:
+            warn(f"The geocentric gravitational constant is negative: '{value}'")
+
+    @property
+    def a(self):
+        "The semimajor axis length [meters]"
+        return self.semimajor_axis
+
+    @property
+    def b(self):
+        "The semimedium axis length [meters]"
+        return self.semimedium_axis
+
+    @property
+    def c(self):
+        "The semiminor axis length [meters]"
+        return self.semiminor_axis
+
+    @property
+    def omega(self):
+        "The angular_velocity [rad s^-1]"
+        return self.angular_velocity
+
+    @property
+    def GM(self):
+        "The Geocentric Gravitational Constant [m^3 s^-2]"
+        return self.geocentric_grav_const
+
+    @property
+    def mean_radius(self):
+        """
+        The arithmetic mean radius :math:`R_1=(a+b+c)/3` [meters]
+        """
+        return (
+            1 / 3 * (self.semimajor_axis + self.semimedium_axis + self.semiminor_axis)
+        )
+
+    @property
+    def volume(self):
+        """
+        The volume of a triaxial ellipsoid :math: `V = /frac{4}{3} pi a b c ` [meters^3]
+        """
+        return (
+            4
+            * np.pi
+            / 3
+            * (self.semimajor_axis * self.semimedium_axis * self.semiminor_axis)
+        )
```

Comment:
Nice! If we add this method on `TriaxialEllipsoid` we should do the same on `Ellipsoid` and `Sphere`, but that's for a separate PR.

Reply:
I have added this as issue #75 so I don't forget!